File: problems/3498-reverse-degree-of-a-string/solution.py
class Solution:
    def reverseDegree(self, s: str) -> int:
        reverse_degree = 0
        for i, char in enumerate(s):
            reverse_alphabet_index = ord('z') - ord(char) + 1
            reverse_degree += reverse_alphabet_index * (i + 1)
        return reverse_degree

File: problems/3498-reverse-degree-of-a-string/solution.java
class Solution {
    public int reverseDegree(String s) {
        int reverseDegree = 0;
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            int alphabetPosition = 26 - (c - 'a');
            reverseDegree += alphabetPosition * (i + 1);
        }
        return reverseDegree;
    }
}